---
title: "gene_info boolean validation"
---

## proteomics

shared data

# Sanity checking booleans after TMT data
How many of these TMT results are significant (pval <= 0.05)
Do they account for these mismatches?
* Component “isAnyProteinChangedInADBrain”: 4141 element mismatches
* Component “protein_brain_change_studied”: 4065 element mismatches
```{r}
sigs <- proteomics_tmt_new[proteomics_tmt_new$cor_pval <= 0.05,]
nrow(sigs)
```

```{r}
# unify protomics & protomics TMT dataframes
prot_concat <- rbind(proteomics_new, proteomics_tmt_new)
prot <- prot_concat$ensembl_gene_id

# find min cor_pval per ENSG across both result sets
prot_mins <- prot_concat %>% group_by(ensembl_gene_id) %>% arrange(cor_pval) %>% slice(1)

# total number of ENSGs we have prot results for (these should have changed == TRUE)
length(unique(prot_concat$ensembl_gene_id))

gene_info <- gene_info_new$ensembl_gene_id
```

1. verify that genes that have no proteomics records are all 'studied == FALSE'
```{r}
prot_not_studied <- setdiff(gene_info, prot) # these ENSGs should have studied == false because no prot records
info_prot_not_studied <- gene_info_new[gene_info_new$ensembl_gene_id %in% prot_not_studied,] 
info_prot_not_studied[info_prot_not_studied$protein_brain_change_studied == TRUE,] # expect empty pre-TMT
```

2. verify that genes that have proteomics records are all 'studied == TRUE'
```{r}
prot_studied <- intersect(gene_info, prot) # these should have studied == true
info_prot_studied <- gene_info_new[gene_info_new$ensembl_gene_id %in% prot_studied,]
info_prot_studied[info_prot_studied$protein_brain_change_studied == FALSE,] # expect empty
```

3. verify that genes with min proteomics cor_pval > 0.05 have 'changed == FALSE'
```{r}
prot_not_changed <- prot_mins[prot_mins$cor_pval > 0.05,]
prot_not_changed_ensgs <- prot_not_changed$ensembl_gene_id
info_prot_not_changed <- gene_info_new[gene_info_new$ensembl_gene_id %in% prot_not_changed_ensgs,]
info_prot_not_changed[info_prot_not_changed$isAnyProteinChangedInADBrain == TRUE,] # expect empty pre TMT
```


4. verify that genes with min proteomics cor_pval <= 0.05 have 'changed == TRUE'
```{r}
prot_changed <- prot_mins[prot_mins$cor_pval <= 0.05,]
prot_changed_ensgs <- prot_changed$ensembl_gene_id
info_prot_changed <- gene_info_new[gene_info_new$ensembl_gene_id %in% prot_changed_ensgs,]
info_prot_changed[info_prot_changed$isAnyProteinChangedInADBrain == FALSE,] # expect empty
```

```{r}
studied <- gene_info_new[gene_info_new$protein_brain_change_studied == TRUE,]
nrow(studied)
```

```{r}
changed <- studied[studied$isAnyProteinChangedInADBrain == TRUE,]
length(unique(changed$ensembl_gene_id))
```

## RNA

shared data

```{r}
rna <- rna_seq_differential_expression_new$ensembl_gene_id
gene_info <- gene_info_new$ensembl_gene_id
rna_mins <- rna_seq_differential_expression_new %>% group_by(ensembl_gene_id) %>% arrange(adj_p_val) %>% slice(1)
```


1. verify that genes that have no rnaseq records are all 'studied == FALSE'
```{r}
rna_not_studied <- setdiff(gene_info, rna) # these should have studied == false
info_rna_not_studied <- gene_info_new[gene_info_new$ensembl_gene_id %in% rna_not_studied,]
info_rna_not_studied[info_rna_not_studied$rna_brain_change_studied == TRUE,] # expect empty
```

2. verify that genes that have rnaseq records are all 'studied == TRUE'
```{r}
rna_studied <- intersect(gene_info, rna) # these should have studied == true
info_rna_studied <- gene_info_new[gene_info_new$ensembl_gene_id %in% rna_studied,]
info_rna_studied[info_rna_studied$rna_brain_change_studied == FALSE,] # expect empty
```

3. verify that genes with min rna adj_p_val > 0.05 have 'changed == FALSE'
```{r}
rna_not_changed <- rna_mins[rna_mins$adj_p_val > 0.05,]
rna_not_changed_ensgs <- rna_not_changed$ensembl_gene_id
info_rna_not_changed <- gene_info_new[gene_info_new$ensembl_gene_id %in% rna_not_changed_ensgs,]
info_rna_not_changed[info_rna_not_changed$isAnyRNAChangedInADBrain == TRUE,] # expect empty
```


4. verify that genes with min proteomics cor_pval <= 0.05 have 'changed == TRUE'
```{r}
rna_changed <- rna_mins[rna_mins$adj_p_val <= 0.05,]
rna_changed_ensgs <- rna_changed$ensembl_gene_id
info_rna_changed <- gene_info_new[gene_info_new$ensembl_gene_id %in% rna_changed_ensgs,]
info_rna_changed[info_rna_changed$isAnyRNAChangedInADBrain == FALSE,] # expect empty
```