---
title: "misc investigations etc."
---
*****
Setup
```{r eval=FALSE, echo=FALSE}
source("~/repos/rstudio/agora-data-validation/src/utils.R")
prepare()
```
*****

## cross file proteomics

```{r}
proteomics_new[proteomics_new$ensembl_gene_id == "ENSG00000178209",]
proteomics_tmt_new[proteomics_tmt_new$ensembl_gene_id == "ENSG00000178209",]

lfq <- proteomics_new$hgnc_symbol
tmt <- proteomics_tmt_new$hgnc_symbol

setdiff(lfq, tmt)
```

## MSBB metadata (adportal thread)
never mind, they shouldn't be joining MSSM to MSBB metadata...

```{r}
individuals <- download_file("syn51942196", type="csv")
metadata <- download_file("syn6101474", type="csv")
```

```{r}
joined <- full_join(individuals, metadata, by="individualID", keep=FALSE)
joined <- joined[,order(colnames(joined))]

interesting_columns <- c("individualID", "Braak.x", "Braak.y", "CERAD.x", "CERAD.y", "diagnosis.x", "CDR")
filtered_joined <- joined[,interesting_columns]
filtered_joined

# diagnosis
unique(individuals$diagnosis)

# mismatched Braak scores
braak_i <- individuals[!is.na(individuals$Braak),]
braak_i

braak_m <- metadata[!is.na(individuals$Braak),]
braak_m

# diagnosis vs CDR scores

```



## gene-biodomain mapping 
What does this look like?
"GO_ID": string
"GOterm_Name": string  
"Biodomain": string    
"ensembl_id": string[]   
"n_ensGene": int    
"entrez_id": string[]   
"n_entrezGene": int 
"symbol": string[]      
"n_symbol": int

```{r}
gene_biodom_mappings <- download_file("syn26529354", type="rds")
colnames(gene_biodom_mappings)
print(gene_biodom_mappings[1])
```


## missing nominated target fields
```{r}
nominations_source <- download_file("syn12540368", type="csv")
```


```{r}
rest <- nominations_source[nominations_source$ensembl_gene_id == 'ENSG00000084093',]
```

```{r}
cols <- colnames(nominations_source)

# no empty strings
nominations_source[nominations_source$Source == "",]
nominations_source[nominations_source$Team == "",]
nominations_source[nominations_source$Rank == "",]
nominations_source[nominations_source$ensembl_gene_id == "",]
nominations_source[nominations_source$hgnc_symbol == "",]
nominations_source[nominations_source$Target_Choice_Justification == "",]
nominations_source[nominations_source$Predicted_Therapeutic_Direction == "",]
nominations_source[nominations_source$input_data == "",] 
nominations_source[nominations_source$initial_nomination == "",]


# empty strings
nominations_source[nominations_source$Data_used_to_support_target_selection == "",] # REST, HDAC1
nominations_source[nominations_source$data_synapseID == "",] # REST, HDAC1, NECTIN2, NGFR
nominations_source[nominations_source$study == "",]  # REST, HDAC1
nominations_source[nominations_source$Validation.study.details == "",] # 104 genes





```




## TMT proteomics source investigation
```{r}
proteomics_source <- download_file("syn18689335", type = "csv")
```


```{r}
tmt_source <- download_file("syn21266454", type = "csv")
```

```{r}
colnames(tmt_source)
```


## TMT vs LFQ uniprot IDs - how aligned are they? (one dropdown or two on details page??)
spot checks on a few disjointed uniprot IDs showed that they have unique proteins, one dropdown should be ok?
```{r}
lfq <- proteomics_old
tmt <-proteomics_tmt_old

# merge(x = df1, y = df2, by = "CustomerId", all = TRUE)
joined <- merge(x = lfq, y = tmt, by = "ensembl_gene_id", all = TRUE)

keeps <- c("ensembl_gene_id", "uniprotid.x", "uniprotid.y", "hgnc_symbol.x", "hgnc_symbol.y")
results <- joined[keeps]
results2 <- results[results$'uniprotid.x' != results$'uniprotid.y',]
results2[complete.cases(results2),]
```

## CI axes for supertable overlay

```{r}
lfq_ci_upr_max <- max(proteomics_new$ci_upr, na.rm=T)
lfq_ci_lwr_min <- min(proteomics_new$ci_lwr, na.rm=T)
tmt_ci_upr_max <- max(proteomics_tmt_new$ci_upr, na.rm=T)
tmt_ci_lwr_min <- min(proteomics_tmt_new$ci_lwr, na.rm=T)
rna_ci_upr_max <- max(rna_seq_differential_expression_new$ci_r, na.rm=T)
rna_ci_lwr_min <- min(rna_seq_differential_expression_new$ci_l, na.rm=T)

cat("lfq ci upper max:", lfq_ci_upr_max, "\nlfq ci lwr min:", lfq_ci_lwr_min,
    "\ntmt ci upper max:", tmt_ci_upr_max, "\ntmt ci lwr min:", tmt_ci_lwr_min,
    "\nrna ci upper max", rna_ci_upr_max, "\nrna ci lwr min:", rna_ci_lwr_min)
```

## Misc

genes with no ME but have DE?

```{r}
t3 <- gene_info_new[gene_info_new$hgnc_symbol == 'TUBAL3',]

gene_info_new[gene_info_new$medianexpression == NULL,]
```




genes with no RNA but Prot?


genes with no RNA but Metab?















```{r}
#stats_file <- download_file("syn27211878", "stats_file", "csv")
#median_file <- download_file("syn12514804", "median_file", "csv")
harmonized_targets <- download_file("syn12540368", "harmonized_targets_file", "csv")
```



how many genes will pull names from RNA DE results instead of geneInfo?
=> 1128



```{r}
blank_hgnc <- gene_info_new[gene_info_new$hgnc_symbol == '',]
ensgs <- blank_hgnc$ensembl_gene_id
rna_seqs <- rna_seq_differential_expression_new[rna_seq_differential_expression_new$ensembl_gene_id %in% ensgs,]
hgnc <- rna_seqs[!rna_seqs$hgnc_symbol == '',]
unique_hgnc <- unique(hgnc$ensembl_gene_id)
print(hgnc[order(hgnc$hgnc_symbol),])
print(unique_hgnc)
```



```{r}
hgnc[hgnc$ensembl_gene_id == 'ENSG00000273036',]
```


Total expected nominations 
```{r}
unique(harmonized_targets$hgnc_symbol)
length(unique(harmonized_targets$hgnc_symbol))
```

Two new nominations
```{r}
ngfr <- harmonized_targets[harmonized_targets$hgnc_symbol == "NGFR",]
nectin2 <- harmonized_targets[harmonized_targets$hgnc_symbol == "NECTIN2",]
print(ngfr)
print(nectin2)
```


```{r}
# find 10 genes that are in both files
# "CBE", "DLPFC", "FP", "IFG", "PHG", "STG", "TCX",

#median_ensgs <- median_file$ensembl_gene_id

  for (i in 1:10) {
    ensg <- median_ensgs[[i]]
    stats <- stats_file[stats_file$ensembl_gene_id == ensg,]
    median <- median_file[median_file$ensembl_gene_id == ensg,]
    
    for (tissue in c("CBE", "DLPFC", "FP", "IFG", "PHG", "STG", "TCX")) {
      tissue_stats <- stats[stats$tissue == tissue,]
      tissue_median <- median[median$tissue == tissue,]
      
      cat("\n", ensg, tissue, "\nStats: ", tissue_stats$medianLogCPM, tissue_stats$meanLogCPM,"\nMedian: ", tissue_median$ medianLogCPM)
      
    }
  }


```



Anna Greenwood (she/her)  12:25 PM
Just new rows. What is the best way to validate that the ENSEMBL ids and gene names are going to be compatable?
12:26
ENSG00000130202	NECTIN2
ENSG00000064300	NGFR

```{r}
gene_info_new[gene_info_new$ensembl_gene_id %in% c('ENSG00000130202', 'ENSG00000064300'),]

```

```{r}
# x is geneInfo, y is proteomics
df1 <- data.frame("ensembl_gene_id" = gene_info_new$ensembl_gene_id, "hgnc_symbol" = gene_info_new$hgnc_symbol)
df2 <- data.frame("ensembl_gene_id" = proteomics_new$ensembl_gene_id, "hgnc_symbol" = proteomics_new$hgnc_symbol)
merge(x = df1, y = df2, by = "ensembl_gene_id", all = TRUE)
```

20 genes from adsp
```{r}


```